src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4296,3 +4296,65 @@ TEST_CASE("TransposedConvolutionWeightInitializationTest", "[ANNLayerTest]")
   REQUIRE(module.Parameters().n_rows
       == (outSize * inSize * kernelWidth * kernelHeight) + outSize);
 }
+
+/**
+ * Simple Test for SpatialDropout layer.
+ */
+TEST_CASE("SpatialDropoutLayerTest", "[ANNLayerTest]")
+{
+  arma::mat input, output, gy, g, outputExpected, gExpected;
+
+  // Set the seed so that the second channel is dropped every time.
+  arma::arma_rng::set_seed(0);
+  SpatialDropout<> module(3, 0.2);
+
+  // Input is a batch of 2 images, each of size (2,2) and having 4 channels.
+  input << 0.4963 << 0.0885 << 0.7682 << 0.1320 << 0.3074 << 0.4901 << 0.6341
+      << 0.8964 << 0.4556 << 0.3489 << 0.6323 << 0.4017 << arma::endr;
```

Comment:
I think the values are from the pytorch implementation?

Reply:
The expected output and expected g matrices are from the Pytorch implementation.  I have added a comment for that a few lines below this.
The input matrix and the gy matrix are filled with random values.

Reply:
Thanks for the clarification.